pkg/filesystem/subvolume.go

```diff
@@ -125,12 +125,27 @@ func getK8sRefSubvolume(ctx context.Context, clientsets *k8sutil.Clientsets) map
 	subvolumeNames := make(map[string]subVolumeInfo)
 	for _, pv := range pvList.Items {
 		if pv.Spec.CSI != nil {
-			subvolumeNames[pv.Spec.CSI.VolumeAttributes["subvolumeName"]] = subVolumeInfo{}
+			subvolumePath := pv.Spec.CSI.VolumeAttributes["subvolumePath"]
+			name := getSubvolumeNameFromPath(subvolumePath)
+			subvolumeNames[name] = subVolumeInfo{}
```

Comment:
if the `name` is empty dont we need to skip it here or log it here?

Reply:
I don't think it is required. If there are no subvolumes, the map will be empty and while lookup all the subvolumes from backend will be considered as stale.

Reply:
why need to have `subvolumeNames[""]=subVolumeInfo{}` this initialization doesnt make sense. we need to log and skip this if we don't find subvolumename

Reply:
oh sorry, got it.

Reply:
need to add continue in `if err!=nil` check to check for other PV

Reply:
added